From 01b639aa7ffbfb87b2fa17cd2953736067e14c30 Mon Sep 17 00:00:00 2001
From: Bai Ping <b51503@freescale.com>
Date: Fri, 22 May 2015 00:36:58 +0800
Subject: [PATCH] MLK-10949 ARM: imx6: add supported wakeup source in soc_pd
 domain for imx6ul

The M/F domain on i.MX6SX is reused on i.MX6UL as SOC_PD domain. The code can be
used by these two SOC. Read the supported wakeup source in SOC_PD domain and add
support for SOC_PD domain on.

Signed-off-by: Bai Ping <b51503@freescale.com>
---
 arch/arm/mach-imx/gpc.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-imx/gpc.c b/arch/arm/mach-imx/gpc.c
index a85ba3e..2815e16 100644
--- a/arch/arm/mach-imx/gpc.c
+++ b/arch/arm/mach-imx/gpc.c
@@ -311,7 +311,7 @@ int imx_gpc_mf_power_on(unsigned int irq, unsigned int on)
 
 int imx_gpc_mf_request_on(unsigned int irq, unsigned int on)
 {
-	if (cpu_is_imx6sx())
+	if (cpu_is_imx6sx() || cpu_is_imx6ul())
 		return imx_gpc_mf_power_on(irq, on);
 	else if (cpu_is_imx7d())
 		return imx_gpcv2_mf_power_on(irq, on);
@@ -334,7 +334,7 @@ void __init imx_gpc_init(void)
 		writel_relaxed(~0, gpc_base + GPC_IMR1 + i * 4);
 
 	/* Read supported wakeup source in M/F domain */
-	if (cpu_is_imx6sx()) {
+	if (cpu_is_imx6sx() || cpu_is_imx6ul()) {
 		of_property_read_u32_index(np, "fsl,mf-mix-wakeup-irq", 0,
 			&gpc_mf_irqs[0]);
 		of_property_read_u32_index(np, "fsl,mf-mix-wakeup-irq", 1,
-- 
1.8.0

